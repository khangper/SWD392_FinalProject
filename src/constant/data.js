export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
