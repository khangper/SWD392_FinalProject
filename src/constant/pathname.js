export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
