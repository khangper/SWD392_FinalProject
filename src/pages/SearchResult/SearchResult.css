.Search_Result_Container {
    width: 100%;
    display: flex;
    padding: 0 370px 0 370px;
    flex-direction: column;
}

.Search_Result_Title {
    display: flex;
    margin-top: 20px;
    flex-direction: column;
}

.Small_SR_Title {
    justify-content: space-between;
    display: flex;
    width: 100%;

}

.Small_SR_Title_Left {
    display: flex;

}

.Small_SR_Title_Right {
    right: 370px;
    display: flex;
    position: absolute;
    font-weight: 400;
    font-style: normal;
    color: rgba(0, 0, 0, .87);
    border: 0.5px solid lightgray;

}

.Small_SR_Title_Right input {

    height: 30px !important;
    font-size: 14px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    border: none;


}

.Small_SR_Title_Right img {
    border-radius: 3px;
    margin-right: 10px;
}

.Main_Title {
    width: 100%;
    padding-bottom: 20px;
    display: inline;
}

.Main_Title h2 {
    font-size: 22px;
    font-weight: 600;
    font-family: 'Roboto', sans-serif;
    margin-top: 70px !important;
    color: #333;
    text-align: left;

}

.Search_Result_Body {
    display: flex;
    width: 100%;
    margin-top: 47px;
    margin-bottom: 5px;
    justify-content: space-between !important;
}

.SR_Left_Col {
    flex: 0 0 auto;
    width: 30%;
}

.Search_Result_Body_Left {
    display: flex;
    padding: 0px 0 15px;
    border-bottom: 1px solid #e5e5e5;

    margin-bottom: 0;
    float: left;
    width: 100%;
    flex-direction: column;
}

.SR_Filter_Sort_Container {
    border-bottom: 1px solid #e5e5e5;
    width: 100%;
    display: flex;
}

.icon_tbl {
    position: absolute;
    left: 38%;
}

.SR_Sort {
    display: flex;
    align-items: center;
    /* padding: 12px; */
    width: 200px;
    color: #333;
    text-decoration: none;
    transition: background-color 0.5s;
    position: absolute;
    left: 36%;
    cursor: pointer;
}

.Sort_Dropdown_Icon {
    width: 11.5px;
    height: 11.5px;
    user-select: none;
    margin-right: 15px;
}

.SR_Dropdown_Sort {
    display: inline-block;
}

.Sort_Menu {
    position: absolute;
    top: 100%;
    left: 0;
    background-color: white;
    box-shadow: 0 8px 16px rgba(0, 0, 0, 0.2);
    z-index: 5;
    list-style: none;
    padding: 0;
    margin: 0;
    width: 100%;
}

.Sort_item {
    padding: 10px 20px;
    cursor: pointer;
}

.Sort_item:hover {
    background-color: #f1f1f1;
}

.SR_Right_Col {
    flex: 0 0 auto;
    width: 75%;
    width: 70%;
    padding-left: 60px;
}

.fcrse_1 {
    background: #fff;
    width: 100%;
    float: left;
    padding: 10px;
    border-radius: 3px;
    border: 1px solid #efefef;
    transition: all .2s ease-in-out;
    display: inline-flex;
}

.hf_img {
    width: 50%;
    float: left;
    position: relative;
}

.hf_img img {
    width: 100%;
    border-radius: 3px;
}

.SR_Course_Overlay {
    position: absolute;
    top: 0;
    padding: 10px;
    left: 0;
    right: 0;
    bottom: 0;
    background-image: linear-gradient(transparent, rgba(51, 51, 51, .3) 90%);
}

.SR_badge_seller {
    border-radius: 3px;
    font-size: 10px;
    position: absolute;
    padding: 2px 10px;
    background: #fa8305;
    border: none;
    color: #fff;
    font-family: 'Roboto', sans-serif;
    font-weight: 600;
    line-height: 1.5;
    right: 0;
    text-align: center;
    text-transform: uppercase;
    top: 10px;
    z-index: 2;
    vertical-align: baseline;
    border-bottom-right-radius: 0;
    border-top-right-radius: 0;
    box-shadow: 0 0 1px 1px rgba(20, 23, 28, .1), 0 3px 1px 0 rgba(20, 23, 28, .1);
}

.SR_crse_reviews {
    color: #fff;
    background: #fdcc0d;
    display: inline-block;
    padding: 3px 15px 3px 10px;
    border-radius: 3px;
    font-size: 14px;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
}

.SR_crse_timer {
    bottom: 0;
    position: absolute;
    margin-bottom: 10px;
    right: 0;
    margin-right: 10px;
    color: #fff;
    background: rgba(51, 51, 51, 0.8);
    display: inline-block;
    padding: 3px 10px;
    border-radius: 3px;
    font-size: 12px;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
}

.SR_play_btn1 {
    border: 0;
    width: 50px;
    height: 50px;
    float: left;
    text-align: center;
    border-radius: 100%;
    padding: 15px 13px;
    font-size: 20px;
    color: #fff;
    position: absolute;
    top: 40%;
    right: 40%;
    opacity: 0;
    background: rgba(51, 51, 51, 0.8);

}

.SR_Content {
    padding: 0px 15px 0px 20px;
    float: left;
    width: 70%;
    position: relative;
}

.More_Icon_DropDown {
    top: 5px;
    display: inline-block;
    position: absolute;
    right: 0;
}

.More_Icon_DropDown a {
    font-size: 20px;
    color: #afafaf !important;
}

.SR_Course_View {
    margin-bottom: 10px;
    font-size: 12px;
    margin-top: 5px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    color: #686f7a;
    text-align: left;
    display: block;
}

.SR_Course_Title {
    display: block;
    font-size: 16px;
    font-size: 14px;
    font-weight: 600;
    font-family: 'Roboto', sans-serif;
    color: #333;
    text-align: left;
    margin-bottom: 7px;
    line-height: 24px;
}

.SR_Course_Sub_Title {
    display: block;
    font-size: 13px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    color: #686f7a;
    text-align: left;
    padding-top: 2px;
}

.SR_Course_Cart {
    float: left;
    width: 100%;
    margin-top: 18px;
}

.SR_Course_Author {
    margin-bottom: 0;
    margin-top: 18px;
    font-size: 14px;
    font-family: 'Roboto', sans-serif;
    float: left;
    color: #686f7a;
    text-align: left;
    line-height: 24px;
}

.SR_Course_Price {
    float: right;
    font-size: 18px;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
    color: #333;
    text-align: right;
}

.SR_cart_btn {}

.MTSR30 {
    margin-top: 30px;
}

.SR_main_loader {
    float: left;
    width: 100%;
    margin-bottom: 20px;
}

.MTSR50 {
    margin-top: 50px;
}

.SR_spinner {
    margin: 0 auto 0;
    width: 50px;
    text-align: center;
    height: 20px;
    border-radius: 100px;
    background-color: #fff;
    line-height: 20px;
    border: 1px solid #efefef;
    cursor: pointer;
}

.SR_spinner .SR_bounce1 {
    -webkit-animation-delay: -0.32s;
    animation-delay: -0.32s;
    background-color: #333;
}

.SR_spinner .SR_bounce2 {
    -webkit-animation-delay: -0.16s;
    animation-delay: -0.16s;
    background-color: #333;
    opacity: 0.702;
}

.SR_spinner .SR_bounce3 {
    -webkit-animation-delay: -0.12s;
    animation-delay: -0.12s;
    background-color: #333;
    opacity: 0.4;
}

.SR_spinner>div {
    width: 10px;
    height: 10px;
    border-radius: 100%;
    display: inline-block;
    /* -webkit-animation: sk-bouncedelay 1.4s infinite ease-in-out both;
    animation: sk-bouncedelay 1.4s infinite ease-in-out both; */
}

.SR_Tbl_Content_Topic h4 {
    display: inline-block;
    /* Để các phần tử nằm trên cùng 1 hàng */
    margin-right: 10px;
    /* Khoảng cách giữa hai phần tử */
}

.SR_Tbl_Content_Topic h4:last-child {
    margin-right: 0;
    /* Loại bỏ khoảng cách bên phải của phần tử cuối cùng */
}

.SR_Tbl_Content_Topic h4:hover {
    cursor: pointer;
    /* Đổi con trỏ chuột khi di chuột qua */
}

.SR_Tbl_Content_Topic_Menu {
    display: none;
    /* Mặc định ẩn đi */
    position: absolute;
    /* Để menu hiển thị đúng vị trí */
    background-color: white;
    /* Nền trắng cho menu */
    box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2);
    /* Đổ bóng cho menu */
    z-index: 1;
    /* Độ cao lớp hiển thị của menu */
}

.SR_Tbl_Content_Topic:hover .SR_Tbl_Content_Topic_Menu {
    display: block;
    /* Hiển thị menu khi di chuột vào */
}

.Topic_item {
    padding: 12px 16px;
    /* Đệm nội dung của mỗi mục */
    text-decoration: none;
    /* Bỏ gạch chân cho các mục */
    display: block;
    /* Để mỗi mục nằm trên một hàng */
}

.Topic_item:hover {
    background-color: #f1f1f1;
    /* Nền xám nhạt khi di chuột qua mục */
    cursor: pointer;
    /* Đổi con trỏ chuột khi di chuột qua */
}

.Sort_Tbl {
    margin-top: 40px;
}