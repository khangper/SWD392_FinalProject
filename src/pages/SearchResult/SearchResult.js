import React, { useState } from 'react';
import './SearchResult.css';
import Footer from '../../components/Footer/Footer';
import Header from '../../components/Header/Header';
import Search from '../../assets/search.png';
import dropdown from "../../assets/dropdown.png";
import ratingStar from "../../assets/rating.png";
import share from "../../assets/share.png";
import saved_course from "../../assets/saved-course.png";
import not_interested from "../../assets/not-interested.png";
import report from "../../assets/report.png";

const SearchResult = () => {
    const [isOpen, setIsOpen] = useState(false);
    const [selectedSort, setSelectedSort] = useState('Sort');
    const [selectedTopic, setSelectedTopic] = useState('Topic');

    const sortOptions = [
        'Most Relevant',
        'Most Reviewed',
        'Highest Rated',
        'Newest',
        'Lowest Price',
        'Highest Price'
    ];

    const toggleDropdown = () => {
        setIsOpen(!isOpen);
    };

    const handleSortClick = (sortOption) => {
        setSelectedSort(sortOption);
        setIsOpen(false);
    };

    const topics = [
        { name: 'SEO', count: 428 },
        { name: 'Php', count: 1526 },
        { name: 'Wordpress Pro', count: 428 },
        { name: 'WooCommerce', count: 958 },
        { name: 'Bootstrap', count: 748 },
        { name: 'Web Development', count: 2256 },
        { name: 'Web Design', count: 4859 },
        { name: 'Digital Marketing', count: 2458 },
        { name: 'E-commerce', count: 1245 },
        { name: 'Wordpress Themes', count: 5879 },
        { name: 'Wordpress Plugins', count: 2654 },
        { name: 'Wordpress Hosting', count: 1485 },
        { name: 'Elementor', count: 3658 },
        { name: 'Wordpress for Ecommerce', count: 5894 }
    ];

    const toggleTopic = () => {
        setIsOpen(!isOpen);
    };

    const handleTopicClick = (topic) => {
        setSelectedTopic(topic.name);
        setIsOpen(false);
    };

    return (
        <>
            <Header />
            <div className='Search_Result_Container'>
                <div className='Search_Result_Title'>
                    <div className='Small_SR_Title'>
                        <div className='Small_SR_Title_Left'>
                            <a href="index.html">Home</a>
                            <p>/ Search Results</p>
                        </div>
                        <div className='Small_SR_Title_Right'>
                            <img src={Search} alt="" className="search-icon" />
                            <input type="text" placeholder="Search " />
                        </div>
                    </div>
                    <div className='Main_Title'>
                        <h2>Search Results</h2>
                    </div>
                </div>
                <div className='Search_Result_Body'>
                    <div className='SR_Left_Col'>
                        <div className='Search_Result_Body_Left'>
                            <div className='SR_Filter_Sort_Container'>
                                <div className="SR_Filter">Filters</div>
                                <div className="SR_Sort" onClick={toggleDropdown}>
                                    <p>{selectedSort}</p>
                                    <span className="SR_Dropdown_Sort">
                                        <img className="Sort_Dropdown_Icon" src={dropdown} alt="Dropdown Icon" />
                                    </span>
                                    {isOpen && (
                                        <div className="Sort_Menu">
                                            {sortOptions.map((option, index) => (
                                                <div
                                                    key={index}
                                                    className="Sort_item"
                                                    onClick={() => handleSortClick(option)}
                                                >
                                                    {option}
                                                </div>
                                            ))}
                                        </div>
                                    )}
                                </div>
                            </div>
                            <div className='Sort_Tbl'>
                                <div className="SR_Tbl_Content">
                                    <div className="SR_Tbl_Content_Topic" onClick={toggleTopic}>
                                        <h4>
                                            {selectedTopic || 'Topic'}
                                        </h4>
                                        <h4 className='icon_tbl'>+</h4>
                                        {isOpen && (
                                            <div className="SR_Tbl_Content_Topic_Menu">
                                                {topics.map((topic, index) => (
                                                    <div
                                                        key={index}
                                                        className="Topic_item"
                                                        onClick={() => handleTopicClick(topic)}
                                                    >
                                                        {topic.name} ({topic.count})
                                                    </div>
                                                ))}
                                            </div>
                                        )}
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                    <div className='SR_Right_Col'>
                        <div className='Search_Result_Body_Right'>
                            <h4 className="SR_Num_Result">5 Results</h4>
                            {[...Array(5)].map((_, index) => (
                                <div key={index} className="fcrse_1 MTSR30">
                                    <a href="https://gambolthemes.net/html-items/cursus-new-demo/course_detail_view.html" className="hf_img">
                                        <img src="https://gambolthemes.net/html-items/cursus-new-demo/images/courses/img-1.jpg" alt="" />
                                        <div className="SR_Course_Overlay">
                                            <div className="SR_badge_seller">Bestseller</div>
                                            <div className="SR_crse_reviews">
                                                <i className="uil uil-star">
                                                    {/* <img className='starIcon' src={ratingStar} alt='rating' /> */}
                                                </i>4.5
                                            </div>
                                            <span className="SR_play_btn1"><i className="uil uil-play"></i></span>
                                            <div className="SR_crse_timer">
                                                25 hours
                                            </div>
                                        </div>
                                    </a>
                                    <div className="SR_Content">
                                        <div className="More_Icon_DropDown">
                                            <a href="#" className="dropdown-button">
                                                â‹®
                                            </a>
                                            <div className="course-more-dropdown-menu">
                                                <span><img src={share} alt='share' />Share</span>
                                                <span><img src={saved_course} alt='saved course' />Save</span>
                                                <span><img src={not_interested} alt='not interested' />Not Interested</span>
                                                <span><img src={report} alt='report' />Report</span>
                                            </div>
                                        </div>
                                        <div className="SR_Course_View">
                                            <span className="SR_Course_View14">109k views</span>
                                            <span className="SR_Course_View14">15 days ago</span>
                                        </div>
                                        <a href="course_detail_view.html" className="SR_Course_Title">Complete Python Bootcamp: Go from zero to hero in Python 3</a>
                                        <a href="#" className="SR_Course_Sub_Title">Web Development | Python</a>
                                        <div className="SR_Course_Cart">
                                            <p className="SR_Course_Author">By <a href="#">John Doe</a></p>
                                            <div className="SR_Course_Price">$10</div>
                                            <button className="SR_cart_btn" title="cart"><i className="uil uil-shopping-cart-alt"></i></button>
                                        </div>
                                    </div>
                                </div>
                            ))}
                        </div>
                    </div>
                </div>
            </div>
            <Footer />
        </>
    );
};

export default SearchResult;
