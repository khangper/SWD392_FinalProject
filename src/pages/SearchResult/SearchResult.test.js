import React from "react";
import { renderWithProviders } from "../../constant/test-util";
import SearchResult from "./SearchResult";
import { screen, waitFor, fireEvent } from "@testing-library/react";

describe("SearchResult", () => {
    beforeAll(() => {
        jest.spyOn(console, "error").mockImplementation(() => { });
        jest.spyOn(console, "log").mockImplementation(() => { });
    });

    afterAll(() => {
        console.error.mockRestore();
        console.log.mockRestore();
    });
    it("renders the SearchResult component", async () => {
        const { container } = renderWithProviders(<SearchResult />);
        expect(container).toBeDefined();
        expect(container).toMatchSnapshot();
    });

});
