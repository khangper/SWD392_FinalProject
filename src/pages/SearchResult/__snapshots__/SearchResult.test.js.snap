// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult renders the SearchResult component 1`] = `
<div>
  <header
    class="header1"
  >
    <a
      href="/home"
    >
      <button
        class="header-link1"
      >
        Back To Cursus
      </button>
    </a>
    <a
      href="/home"
    >
      <img
        alt=""
        class="header-group-icon"
        src="https://gambolthemes.net/html-items/cursus-new-demo/images/logo.svg"
      />
    </a>
    <img
      alt=""
      class="header-avatar"
      src="profile-img.jpg"
    />
  </header>
  <div
    class="Search_Result_Container"
  >
    <div
      class="Search_Result_Title"
    >
      <div
        class="Small_SR_Title"
      >
        <div
          class="Small_SR_Title_Left"
        >
          <a
            href="/home"
          >
            Home 
          </a>
          <p>
            / Search Results
          </p>
        </div>
        <div
          class="Small_SR_Title_Right"
        >
          <img
            alt=""
            class="search-icon"
            src="search.png"
          />
          <input
            placeholder="Search "
            type="text"
          />
        </div>
      </div>
      <div
        class="Main_Title"
      >
        <h2>
          Search Results
        </h2>
      </div>
    </div>
    <div
      class="Search_Result_Body"
    >
      <div
        class="SR_Left_Col"
      >
        <div
          class="Search_Result_Body_Left"
        >
          <div
            class="SR_Filter_Sort_Container"
          >
            <div
              class="SR_Filter"
            >
              Filters
            </div>
            <div
              class="SR_Sort"
            >
              <p>
                Sort
              </p>
              <span
                class="SR_Dropdown_Sort"
              >
                <img
                  alt="Dropdown Icon"
                  class="Sort_Dropdown_Icon"
                  src="dropdown.png"
                />
              </span>
            </div>
          </div>
          <div
            class="Sort_Tbl"
          >
            <div
              class="SR_Tbl_Content"
            >
              <div>
                <span>
                  Topic
                </span>
                <span>
                  +
                </span>
              </div>
            </div>
            <div
              class="SR_Tbl_Content"
            >
              <div>
                <span>
                  Levels
                </span>
                <span>
                  +
                </span>
              </div>
            </div>
            <div
              class="SR_Tbl_Content"
            >
              <div>
                <span>
                  Languages
                </span>
                <span>
                  +
                </span>
              </div>
            </div>
            <div
              class="SR_Tbl_Content"
            >
              <div>
                <span>
                  Prices
                </span>
                <span>
                  +
                </span>
              </div>
            </div>
            <div
              class="SR_Tbl_Content"
            >
              <div>
                <span>
                  Features
                </span>
                <span>
                  +
                </span>
              </div>
            </div>
            <div
              class="SR_Tbl_Content"
            >
              <div>
                <span>
                  Ratings
                </span>
                <span>
                  +
                </span>
              </div>
            </div>
            <div
              class="SR_Tbl_Content"
            >
              <div>
                <span>
                  Video Durations
                </span>
                <span>
                  +
                </span>
              </div>
            </div>
            <div
              class="SR_Tbl_Content"
            >
              <div>
                <span>
                  Close Captions
                </span>
                <span>
                  +
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="SR_Right_Col"
      >
        <div
          class="Search_Result_Body_Right"
        >
          <h4
            class="SR_Num_Result"
          >
            4
             Results
          </h4>
        </div>
      </div>
    </div>
    <section
      class="load-more"
    >
      <div
        class="spinner"
      >
        <div
          class="bounce1"
        />
        <div
          class="bounce2"
        />
        <div
          class="bounce3"
        />
      </div>
    </section>
  </div>
  <footer
    class="footer footer-large-container"
  >
    <div
      class="footer-links"
    >
      <div
        class="footer-column"
      >
        <a
          class="link"
          href="/about-us"
        >
          About
        </a>
        <a
          class="link"
          href="/blog-single-view/:id"
        >
          Blog
        </a>
        <a
          class="link"
          href="/career"
        >
          Careers
        </a>
        <a
          class="link"
          href="/press"
        >
          Press
        </a>
      </div>
      <div
        class="footer-column"
      >
        <a
          class="link"
          href="/help"
        >
          Help
        </a>
        <a
          class="link"
          href="/coming-soon"
        >
          Advertise
        </a>
        <a
          class="link"
          href="/coming-soon"
        >
          Developers
        </a>
        <a
          class="link"
          href="/contactus"
        >
          Contact Us
        </a>
      </div>
      <div
        class="footer-column"
      >
        <a
          class="link"
          href="/termsofuse"
        >
          Copyright Policy
        </a>
        <a
          class="link"
          href="/termsofuse"
        >
          Terms
        </a>
        <a
          class="link"
          href="/termsofuse"
        >
          Privacy Policy
        </a>
        <a
          class="link"
          href="/sitemap"
        >
          Sitemap
        </a>
      </div>
      <div
        class="footer-column"
      >
        <button
          class="teach-button"
        >
          Teach on Cursus
        </button>
        <div
          class="language-selector"
        >
          <button>
            Language 
            <span>
              ▼
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="footer-bottom"
    >
      <div
        class="footer-copy"
      >
        <img
          alt=""
          src="https://gambolthemes.net/html-items/cursus-new-demo/images/logo1.svg"
        />
        © 2024 
        <strong>
          Cursus
        </strong>
        . All Rights Reserved.
      </div>
      <div
        class="footer-social"
      >
        <a
          href="/home"
        >
          <img
            alt=""
            class="social-icon"
            src="facebook.png"
          />
        </a>
        <a
          href="/home"
        >
          <img
            alt=""
            class="social-icon"
            src="x.png"
          />
        </a>
        <a
          href="/home"
        >
          <img
            alt=""
            class="social-icon"
            src="google+.png"
          />
        </a>
        <a
          href="/home"
        >
          <img
            alt=""
            class="social-icon"
            src="linkedin.png"
          />
        </a>
        <a
          href="/home"
        >
          <img
            alt=""
            class="social-icon"
            src="insta.png"
          />
        </a>
        <a
          href="/home"
        >
          <img
            alt=""
            class="social-icon"
            src="pinterest.png"
          />
        </a>
      </div>
    </div>
  </footer>
</div>
`;
